src/pages/AppDetails/AppDetailsPage.tsx

```diff
@@ -113,12 +113,14 @@ class AppDetails extends React.Component<RouteComponentProps<AppId>, AppDetailsS
               <NavItem eventKey="out_metrics">
                 <div>Outbound Metrics</div>
               </NavItem>
-              {this.state.app.customDashboards.map(dashboard => {
-                return (
-                  <NavItem key={dashboard.template} eventKey={dashboard.template}>
-                    <div>{dashboard.title}</div>
-                  </NavItem>
-                );
+              {this.state.app.runtimes.map(runtime => {
+                return runtime.dashboardRefs.map(dashboard => {
+                  return (
+                    <NavItem key={dashboard.template} eventKey={dashboard.template}>
```

Comment:
Minor, if you have two runtimes with same template i.e. Vert.x - Client, EAP - Client, this could throw a warning.
Perhaps better to add runtime id in the key.

Reply:
Hmm I don't think, as the template here refers to the dashboard name, so it is unique (at least in a namespace).